gobblin-runtime/src/main/java/org/apache/gobblin/runtime/kafka/HighLevelConsumer.java

```diff
@@ -348,13 +348,15 @@ record = queue.take();
             // Committed offset should always be the offset of the next record to be read (hence +1)
             partitionOffsetsToCommit.put(partition, record.getOffset() + 1);
           }
+        } catch (InterruptedException e) {
+          // Stop queue processor and return when encountered InterruptedException
+          log.warn("Thread interrupted while processing queue ", e);
+          Thread.currentThread().interrupt();
+          return;
+        } catch (Exception e) {
+          // Log the error and let the queue processor continue processing
+          log.error("Encountered exception while processing record. Record: {} Exception: {}", record, e);
         }
```

Comment:
I agree that previously exiting `Runnable.run` upon `Exception` was surprising and subtle and generally not helpful, but this alternative could just put us into a continuous loop where we might encounter the same error "forever".  should we cap the number of retries using something like a `Retryer`, as [here](https://github.com/apache/gobblin/blob/45ad13e07ebe212002b279132e1fa4d121a9294f/gobblin-service/src/main/java/org/apache/gobblin/service/monitoring/KafkaJobStatusMonitor.java#L123)?

Reply:
Actually there is no retry happening here during ProcessMessage, are you suggesting we implement a retry mechanism here?

Reply:
yes, exactly.  right now, it's an infinite loop w/ no recovery (so obviously no retries).

this PR would change that infinite loop to do infinite recovery, although NEVER retry a given message (any msg hitting an error gets skipped).

I'm suggesting to change the infinite loop to do possibly infinite recovery with bounded retries on any given message.  recovery only continues as long as msg retries resolve within bounded time.

I'd let retries continue a configurable time, probably up to 15 mins or so in our case of gobblin cluster.  at that point fail loudly via a metrics condition to alert on.  (I'd set threshold very low there - like just one event.)

Reply:
I think we can keep the behaviour same in case of manual commit. Since in those cases commit only happens when the message is processed.

But in case of autoCommit, messages are committed as soon as they are consumed. Even when the `processQueue` infinite loop is broken, messages are consumed and committed. In such scenarios it would be better to let future messages be processed if they could.

Reply:
Also, adding a retry may cause some hidden effects like delay in message processing instead of an early failure. It may also introduce delays in catching a true failure.

Since this piece is getting used at a lot of places, I would like to keep the change minimal.